client.go

```diff
@@ -491,6 +491,7 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 		uniqueInserter: baseservice.Init(archetype, &dbunique.UniqueInserter{
 			AdvisoryLockPrefix: config.AdvisoryLockPrefix,
 		}),
+		workCancel: func(cause error) {}, // replaced on start, but here in case StopAndCancel is called before start up
```

Comment:
for such usecase I would define a `defautWorkCanceler := func(error) {}`

Then use it here, it helps to document separately, and using 


```suggestion
		workCancel: defautWorkCanceler // replaced on start
```

Reply:
Hm, not sure I agree on that one :) You're never going to reuse this function, and I think adding a named variable for it introduces a layer of indirection that in the end just serves to make it less clear what the code's doing.